companies/views.py

```diff
@@ -75,26 +75,94 @@ def perform_create(self, serializer):
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request, startup_id):
+        """Allow an investor to follow a startup."""
+        investor_company = self.get_investor_company(request)
+        startup = self.get_startup(startup_id)
+
         if CompanyFollowers.objects.filter(investor=investor_company, startup=startup).exists():
             return Response({"detail": "You are already following this startup."}, status=status.HTTP_400_BAD_REQUEST)
 
         follow_relation = CompanyFollowers.objects.create(investor=investor_company, startup=startup)
         serializer = CompanyFollowersSerializer(follow_relation)
 
-        return Response(serializer.data, status=status.HTTP_201_CREATED)
\ No newline at end of file
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+    
+class UnFollowStartupView(APIView, InvestorStartupMixin):
+    """A view that allows an investor to unfollow a startup."""
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request, startup_id):
+        """
+        Parameters:
+        startup_id (int): The ID of the startup to unfollow.
+
+        Response: 
+        A success message if unfollowed, or an error message.
+        """
+        investor_company = self.get_investor_company(request)
+        startup = self.get_startup(startup_id)
+
+        unfollow_relation = CompanyFollowers.objects.filter(investor=investor_company, startup=startup)
+        if not unfollow_relation.exists():
+            return Response({"detail": "You are not following this startup."}, status=status.HTTP_400_BAD_REQUEST)
+
+        unfollow_relation.delete()
+
+        return Response({"detail": "Successfully unfollowed the startup."}, status=status.HTTP_200_OK)
+    
+class ListFollowedStartupsView(APIView, InvestorStartupMixin):
+    """
+    A view that lists all followed startups by an investor
+    """
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request):
+        """
+        Responses:
+        - 200 OK: Returns a list of followed startups.
+        - 400 Bad Request: If the user is not linked to any company or is not an enterprise.
+
+        Example Request:
+        GET /api/investor/saved-startups?search=tech&order_by=-created_at
+        """
+        investor_company = self.get_investor_company(request)
+        
+        startups = CompanyProfile.objects.filter(
+            startup_investors__investor=investor_company
+        )
+        search_query = request.query_params.get("search", None)
+        order = request.query_params.get("order_by", "company_name")
+
+        if search_query:
+            startups = startups.filter(company_name__icontains=search_query)
+
+        allowed_order_fields = ["company_name", "created_at", "description", "updated_at"]
+        if order.lstrip("-") in allowed_order_fields:
+            startups = startups.order_by(order)
+
+        serializer = CompanyProfileSerializer(startups, many=True)
```

Comment:
Using `CompanyProfileSerializer` might expose unnecessary fields. Create a slimmed-down serializer for followed startups, if needed:
```
class FollowedStartupSerializer(serializers.ModelSerializer):
    class Meta:
        model = CompanyProfile
        fields = ["id", "company_name", "description", "website"]
```